core/src/main/scala/caliban/Rendering.scala

```diff
@@ -70,8 +75,8 @@ object Rendering {
 
   private[caliban] def renderTypeName(fieldType: __Type): String =
     fieldType.kind match {
-      case __TypeKind.NON_NULL => s"${fieldType.ofType.map(renderTypeName).getOrElse("null")}!"
-      case __TypeKind.LIST     => s"[${fieldType.ofType.map(renderTypeName).getOrElse("null")}]"
+      case __TypeKind.NON_NULL => s"${fieldType.ofType.fold("null")(renderTypeName)}!"
+      case __TypeKind.LIST     => s"[${fieldType.ofType.fold("null")(renderTypeName)}]"
```

Comment:
we should merge the two cases into one here with a or of the two possible case values yielding a common outcome.